### ISSUE TITLE
Formatting option to insert newline after file scoped namespace declarations

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#67400]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Formatting-option-to-insert-newline-afte/10307275)._

---
Currently, the default formatter for C# does not care whether you have a newline after a file scoped namespace or not.

```
// This:

namespace OL. PharmaPos.Modules.ProductsCatalog.Domain.Products;
using FluentResults;

public readonly record struct ProductId
{
}

// ... as well as this:

namespace OL. PharmaPos.Modules.ProductsCatalog.Domain.Products;

using FluentResults;

public readonly record struct ProductId
{
}

// ... are perfectly accepted by VS 2022, currently.
```

I would like to have the option to make the first variant illegal or be automatically formatted on document format (Ctrl+K, Ctrl+D) to look like the second variant.
Basically this inserts a new line after a file scoped namespace declaration.

---
#### Feedback Bot on 3/13/2023, 01:18 AM: 

(private comment, text removed)

**Comment by nazarkryp** [comment]
why is this ignored? to busy adding additional "very useful" item to project' right-mouse click context menu?

**Comment by sagarok** [comment]
Is it going to be implemented?

**Comment by Inurias** [comment]
Hi @sharwell and @CyrusNajmabadi, I'd be interested in giving this a shot. However, I'm not very familiar with the Roslyn codebase. 

Should I add the new required functionality to the `NewLineUserSettingFormattingRule` or add a new rule for this?
Or should this be done somewhere else entirely?
Where would the new test for this go?

> I don't think we need an option here, since we could simply update the formatter to always apply the blank line.

I think this is a good idea, but wouldn't this still require the user to apply the formatter at least once always - since newly created classes that are filled with the default declaration text will be created without the extra blank line inbetween?

**Comment by CyrusNajmabadi** [comment]
>  since newly created classes that are filled with the default declaration text will be created without the extra blank line inbetween?

Newly created classes go through the formatting engine. :)

> Should I add the new required functionality to the NewLineUserSettingFormattingRule

No.  That relates to options controlled by a user setting.  We just want to have the standard formatting rule handle this.   I believe that is just TokenBasedFormattingRule, though i could be wrong.

**Comment by Inurias** [comment]
Fixed by https://github.com/dotnet/roslyn/pull/77821
